### PR TITLE
CLM AppStream filters: module picker controls

### PR DESCRIPTION
**Pull request by cbbayburt**
Add module picker controls for CLM AppStreams filter form.

## GUI diff
Before:
combo-before

After (in order of flow):
combo-browse
combo-channel
combo-module
combo-stream

## Documentation
- https://github.com/SUSE/spacewalk/issues/11486

## Test coverage
- No tests: Not testable

## Links

Fixes https://github.com/SUSE/spacewalk/issues/10097

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"  
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/suse/manager/webui/controllers/contentmanagement/handlers/AppStreamsApiController.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     * @return the JSON data
+     */
+    public static String getModulesInChannel(Request req, Response res, User user) {
+        try {
+            Channel channel = ChannelManager.lookupByIdAndUser(Long.parseLong(req.params("channelId")), user);
+            return json(res, ResultJson.success(api.getAllModulesInChannel(channel)));
+        }
+        catch (NumberFormatException e) {
+            Spark.halt(HttpStatus.SC_BAD_REQUEST);
+            return null;
```
Comment (hustodemon) [inline java/code/src/com/suse/manager/webui/controllers/contentmanagement/handlers/AppStreamsApiController.java:61]: Nit: is the return needed? AFAIK, `Spark.halt` throws an exception.

Comment (cbosdo) [reply java/code/src/com/suse/manager/webui/controllers/contentmanagement/handlers/AppStreamsApiController.java:61]: Agreed, this is not needed

File: java/spacewalk-java.changes [hunk 2]
```diff
@@ -1,3 +1,4 @@
+- Implement module picker controls for CLM AppStream filters
```
Comment (cbosdo) [inline java/spacewalk-java.changes:1]: is CLM an acronym the customers know of? I would rather have it expanded even if that make a long line.

File: web/html/src/manager/content-management/list-filters/appstreams/appstreams.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        setLoading(false);
+        setBrowse(true);
+      }).catch(xhr => showErrorToastr(Network.responseErrorMessage(xhr)));
+  }
+
+  return (
+    isBrowse ?
+      <SelectInput channels={channels}/>
+    :
+      <React.Fragment>
```
Comment (cbosdo) [inline web/html/src/manager/content-management/list-filters/appstreams/appstreams.js:28]: You can replace `<React.Fragment></React.Fragment>` by this short version `<></>`

File: java/spacewalk-java.changes (hunk 2 above)
Comment (cbbayburt) [reply java/spacewalk-java.changes:1]: Yes, "CLM" acronym is also mentioned in the docs, so I'd say it's safe to use it here as well

**Comment by cbbayburt** [comment]
@hustodemon, @cbosdo thanks for the reviews. I'll merge it as soon as the branch is free.

**Comment by cbbayburt** [comment]
Thanks, @ncounter!

> There is just one tiny remark: now there are two `<select>` boxes with different styles in the same form/page, they look a little bit weird. Unless you know how to fix it in 10 minutes, let's keep it as it is, it is fine for the context of this PR, never mind.

I agree. However, I spent some time looking into it, but branding the combo box needs a bit more dedication, so maybe we can cover it in a separate issue.

**Pull request merged**